### PR TITLE
creditential login and signup

### PR DESCRIPTION
**Pull request by phoenixdahdev**
# Description

**Closes #issue_number_here**

# Changes proposed

## What did you do?

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [ ] My code follows the code style of this project.
- [ ] This PR does not contain plagiarized content.
- [ ] The title and description of the PR is clear and explains the approach.
- [ ] I am making a pull request against the **dev branch** (left side).
- [ ] My commit messages styles matches our requested structure.
- [ ] My code additions will fail neither code linting checks nor unit test.
- [ ] I am only making changes to files I was requested to.

# Screenshots/Videos

### CODE REVIEW

File: src/actions/login.ts
Comment (prudentbird) [inline src/actions/login.ts:1]: Send file to utils

All endpoints are suppose to return the same response

File: package.json
Comment (prudentbird) [inline package.json:1]: Kindly undo package updates abeg

File: src/actions/register.ts
Comment (prudentbird) [inline src/actions/register.ts:1]: Looks good

File: src/config/auth.config.ts [hunk 1]
```diff
@@ -50,39 +64,58 @@ export default {
...
       )) as ApiResponse;
 
       user = response?.data?.user;
 
-      return { ...token, user };
+      return { ...token, ...user };
     },
-    async session({ session }) {
+    async session({ session, token }) {
+      // @ts-expect-error setting user on login from all response
```
Comment (prudentbird) [inline src/config/auth.config.ts:106]: Don't disable lint, fix the lint instead

File: src/config/auth.config.ts [hunk 2]
```diff
@@ -50,39 +64,58 @@ export default {
...
     },
-    async redirect({ url, baseUrl }) {
-      if (url === "/login") {
-        return baseUrl;
-      }
-      if (url === `${baseUrl}/api/auth/signout`) {
-        return baseUrl;
-      }
-      return "/dashboard/admin";
-    },
```
Comment (prudentbird) [inline src/config/auth.config.ts:85]: Add the redirect, and return "/dashboard", not admin

File: src/lib/auth.ts [hunk 3]
```diff
@@ -1,7 +1,18 @@
-import NextAuth from "next-auth";
+import NextAuth, { type DefaultSession } from "next-auth";
 
 import authConfig from "~/config/auth.config";
 
+interface User {
+  id: string;
```
Comment (prudentbird) [inline src/lib/auth.ts:6]: Extract all type declarations to ~/types across your files

File: src/schemas/index.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import * as z from "zod";
```
Comment (prudentbird) [inline src/schemas/index.ts:1]: Beautiful schema!

**Review by prudentbird — CHANGES_REQUESTED** [review]

**Review by prudentbird — COMMENTED** [review]
Make changes, submit another PR

File: package.json
Comment (phoenixdahdev) [reply package.json:1]: aiit

**Pull request closed without merging**